int32 input1
int32 input2
int32 input3
int32 input4
